core/src/main/java/org/mule/runtime/core/internal/execution/SourceResultAdapter.java

```diff
@@ -127,6 +132,15 @@ public PayloadMediaTypeResolver getPayloadMediaTypeResolver() {
     return payloadMediaTypeResolver;
   }
 
+  /**
+   * @return information about the polled item that originated the message
```

Comment:
bad javadoc. It doesn't explain that this would only be available when the message has been generated by a PollingSource